addons/purchase/tests/test_purchase.py

```diff
@@ -942,3 +942,54 @@ def test_vendor_price_by_purchase_order_company(self):
         self.assertEqual(po.amount_untaxed, 15.0)
         po.company_id = company_a.id
         self.assertEqual(po.amount_untaxed, 10.0)
+
+    def test_bill_in_purchase_matching_individual(self):
+        """
+        Tests that if the vendor is an individual with a company, the bill will still appear when
+        using the purchase matching button on a vendor bill
+        """
+        company_partner = self.env['res.partner'].create({
+            'name': 'Small Company',
+            'company_type': 'company',
+        })
+        individual_partner = self.env['res.partner'].create({
+            'name': 'Big Vendor',
+            'company_type': 'person',
+            'parent_id': company_partner.id,
+        })
```

Comment:
Use existing data when possible 

```suggestion
       self.partner_a.parent_id = company_partner
```